x-pack/plugins/observability_solution/inventory/public/hooks/use_entity_definition.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useAbortableAsync } from '@kbn/observability-utils-browser/hooks/use_abortable_async';
+import { useKibana } from './use_kibana';
+
+export const useEntityDefinition = (id: string) => {
+  const {
+    services: { entityManager },
+  } = useKibana();
+
+  const { value, loading, refresh } = useAbortableAsync(
+    ({ signal }) => {
+      return entityManager.entityClient.getEntityDefinition(id);
+    },
+    [entityManager.entityClient, id]
+  );
+
+  return {
+    entityDefinitions: value?.definitions,
+    isEnablementLoading: loading,
+    refresh,
```

Comment:
Does it make sense to expose a refresh here?

Reply:
Removed, no need to, thanks!